### PR TITLE
Introduce ViewCoordinates archetype

### PR DESCRIPTION
**Pull request by jleibs**
### What
New type definitions, tests, and round-trips for ViewCoordinates across all 3 languages.

This introduces another small script generator to create constants for all the permutations of ViewCoordinates and add them to each of the archetypes.

Not in this PR:
 - Migrating to the new Archetype in the viewer
 - Introducing ViewCoordinate code-examples (we didn't have existing examples, and this requires a bit of thought for how to show off)

Part of #2816

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3390) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3390)
- [Docs preview](https://rerun.io/preview/3adda06aabcbad19850ed49f0645ea36ac26579e/docs) 
- [Examples preview](https://rerun.io/preview/3adda06aabcbad19850ed49f0645ea36ac26579e/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: rerun_py/rerun_sdk/rerun/components/view_coordinates_ext.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+        if data.shape != (3,):
+            raise ValueError(f"ViewCoordinates must be a 3-element array. Got: {data.shape}")
+
+        for value in data:
+            # TODO(jleibs): Enforce this validation based on ViewDir
+            if value not in range(1, 7):
+                raise ValueError("ViewCoordinates must contain only values in the range [1,6].")
+
+        return pa.FixedSizeListArray.from_arrays(data, type=data_type)
```
Comment (teh-cmc) [inline rerun_py/rerun_sdk/rerun/components/view_coordinates_ext.py:40]: Unless I'm misreading this, this feels wrong to me.
The component has no idea whether it should be mono or not: that's a property of how the archetype was defined.
If there happens to be multiple instances of ViewCoordinates, then this should serialize all of them without question.

What if something tries to define a custom archetype that actually uses a batch, or even just logs a batch directly using the low level APIs?

File: scripts/generate_view_coordinate_defs.py
Comment (teh-cmc) [inline scripts/generate_view_coordinate_defs.py:1]: `TODO(#2388): remove this script`

**Review by teh-cmc — APPROVED** [review]

File: crates/re_types/definitions/rerun/archetypes/view_coordinates.fbs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+include "rerun/datatypes.fbs";
+include "rerun/components.fbs";
+
+namespace rerun.archetypes;
+
+
+/// How we interpret the coordinate system of an entity/space.
+table ViewCoordinates (
+  "attr.rust.derive": "Copy, Eq, PartialEq",
```
[the remark this answers is not in the document]
Comment (jleibs) [reply crates/re_types/definitions/rerun/archetypes/view_coordinates.fbs:11]: Is this the exception to the "always sort all the things" rule?

File: rerun_py/rerun_sdk/rerun/components/view_coordinates_ext.py (hunk 1 above)
Comment (jleibs) [reply rerun_py/rerun_sdk/rerun/components/view_coordinates_ext.py:40]: Ah yeah, you're right. Good catch.

**Pull request merged**